lib/model/budget.dart

```diff
@@ -1,12 +1,16 @@
-import 'package:sossoldi/model/base_entity.dart';
+import '../database/sossoldi_database.dart';
+import 'base_entity.dart';
 
-const String budget = 'budget';
+const String budgetTable = 'budget';
 
 class BudgetFields extends BaseEntityFields {
+  static String id = 'id';
   static String name = 'name';
   static String amountLimit = 'amountLimit';
+  static String createdAt = 'createdAt';
+  static String updatedAt = 'updatedAt';
```

Comment:
The fields `id`, `createdAt` and `updateAt` are in the abstract class `BaseEntiryFileds`, why did you add them in this concrete class?

Reply:
`BaseEntityFields` is an _abstract_ class, so its properties MUST be declared in extended class, but they are not automatically inherited. If you remove line 7, you will get "_The getter 'id' isn't defined for the type 'BudgetFields'._" error on `_createDB()` func.

BTW there's a error anyway here, on lines 7, 10, and 11 that should be:
`  static String id = BaseEntityFields.getId;
  static String name = 'name';
  static String amountLimit = 'amountLimit';
  static String createdAt = BaseEntityFields.getCreatedAt;
  static String updatedAt = BaseEntityFields.getUpdatedAt;`
  for all models. I will edit they.

Reply:
Usually a field declared in the abstract class if it is public, the concrete class can use it without problems. This in practically any object oriented programming language. Maybe in dart it works differently...